File: library/src/main/java/com/firemaples/googlewebtranslator/LogTool.java
package com.firemaples.googlewebtranslator;

import android.util.Log;

public class LogTool {
    public static boolean _debug = true;

    public static void setDebug(boolean debug) {
        _debug = debug;
    }

    public static boolean isDebug() {
        return _debug;
    }

    public static void d(String tag, String msg) {
        if (!_debug) {
            return;
        }

        Log.d(tag, msg);
    }

    public static void i(String tag, String msg) {
        if (!_debug) {
            return;
        }

        Log.i(tag, msg);
    }

    public static void w(String tag, String msg) {
        if (!_debug) {
            return;
        }

        Log.w(tag, msg);
    }

    public static void e(String tag, String msg) {
        if (!_debug) {
            return;
        }

        Log.e(tag, msg);
    }
}

File: library/src/main/java/com/firemaples/googlewebtranslator/LogTool.kt
package com.firemaples.googlewebtranslator

import android.util.Log

class LogTool {
    companion object {
        @JvmStatic
        fun d(tag: String, msg: String, force: Boolean = false) {
            if (!Tools.DEBUG && !force) {
                return
            }

            Log.d(tag, msg)
        }

        @JvmStatic
        fun i(tag: String, msg: String, force: Boolean = false) {
            if (!Tools.DEBUG && !force) {
                return
            }

            Log.i(tag, msg)
        }

        @JvmStatic
        fun w(tag: String, msg: String, force: Boolean = false) {
            if (!Tools.DEBUG && !force) {
                return
            }

            Log.w(tag, msg)
        }

        @JvmStatic
        fun e(tag: String, msg: String, force: Boolean = false) {
            if (!Tools.DEBUG && !force) {
                return
            }

            Log.e(tag, msg)
        }
    }
}
